### ISSUE TITLE
Clarification: JacVecOperator interface

### ISSUE THREAD

**Issue by lindnemi** [SciML/DiffEqOperators.jl#328]
Hi, I'm trying to implement a custom JacVecOperator.

In the [docs](https://diffeq.sciml.ai/stable/features/diffeq_operator/#AbstractDiffEqOperator-Interface-Description) it sounds like L(u,p,t) should be multiplication of L with a vector `u`. If `L = Jac(x)` is some Jacobian at location `x` that would be the directional derivative `Jac(x) * u`.

On the other hand [here](https://github.com/SciML/DiffEqOperators.jl/blob/97c6b24943960d1a4d972fcbca4e5c13347bcef4/src/jacvec_operators.jl#L113) it seems that the function call `L(u,p,t)` will always evaluate to `Jac(u) * u`. Meaning that the direction and the point at which Jac is evaluated will always be the same.

In this case, how am I supposed to implement the general directional derivative `Jac(u) * z` where `z` is an arbitrary direction?

**Comment by ChrisRackauckas** [comment]
`*` should not update the Jacobian, as you say that would allow `Jac(x) * u`. Using `L(u,p,t)` as a function inside of the ODE defines the ODE `Jac(u)*u` which is a representation of the original nonlinear operator and is thus useful in things like local sensitivity analysis, hence the default.

**Comment by lindnemi** [comment]
Thank you for the quick answer. So there are basically three things to implement for a JacVecOperator:

- `L(u,p,t)` for Jac(u) * u
- `*` for Jac(x) * u
- `update_coefficients`

Each of which should be given in mutating and oop form. I would love to see a brief section specifically about JacVecOperator in the docs.

**Comment by ChrisRackauckas** [comment]
Yes, http://diffeqoperators.sciml.ai/dev/operators/jacobian_vector_product/ needs a bit more.
